app/Http/Controllers/LoginController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers;
+
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response as HttpResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Response;
+
+class LoginController extends Controller
+{
+  /**
+   * Display a listing of the resource.
+   */
+
+
+
+  public function index(Request $request)
+  {
+    //
+  }
+
+  /**
+   * Store a newly created resource in storage.
+   */
+  public function authenticate(Request $request)
+  {
+
+    $credentials = $request->validate([
+      'email' => ['required', 'email'],
+      'password' => ['required'],
+    ]);
```

Comment:
Tout ce qui est validateur d'un controller tu peux le mettre dans une form request : https://laravel.com/docs/10.x/validation#form-request-validation

Ca délocaliser la responsabilité

Reply:
Created class AuthenticateUserRequest that somehow makes a magic validation. Validation is working. Had to declare `authorize()` to `true `to make it work

Reply:
Same approach is used in StoreRegisterRequest